rcl/src/rcl/node.c

```diff
@@ -320,6 +321,8 @@ rcl_node_init(
   node->impl->rmw_node_handle = rmw_create_node(
     &(node->context->impl->rmw_context),
     name, local_namespace_, domain_id, &node_security_options);
+  // node_secure_root is only used for rmw_create_node
+  allocator->deallocate(node_secure_root, allocator->state);
```

Comment:
Might need to `node_secure_root = NULL` to avoid double free in `cleanup`

Reply:
line 311 has an if-else loop that checks the value of `node_secure_root`. I think when it gets to this point, `node_secure_root` will not be NULL.

Reply:
I was trying to say `node_secure_root` needs to be set to `NULL` after deallocating. On second thought, this deallocate isn't needed because [line 360](https://github.com/ros2/rcl/blob/214b23bea0cdde93244bbdbb53d6d29358ef0922/rcl/src/rcl/node.c#L360) jumps to `cleanup` before the function returns.

Reply:
We should probably always reset the pointer to NULL anyway to be on the safe side...

Reply:
@sloretz thanks for pointing out. removed.